frontend/modules/recipe_form/components/RecipeForm.js

```diff
@@ -50,7 +53,15 @@ class RecipeForm extends React.Component {
   }
 
   _onInit() {
-    this.setState( this.getStateFromStore() );
+    const state = this.getStateFromStore();
+    this.setState(state);
+
+    if (Object.keys(state.data).length > 0) {
+      const user = this.getAuthUser()
+      if (state.data.author !== user.id) {
+        browserHistory.push('/');
```

Comment:
The second point brings up the issue #65, we should be doing basic auth checks before the user even hits that path.